application/src/main/java/org/opentripplanner/model/TimetableSnapshot.java

```diff
@@ -552,7 +552,39 @@ public boolean isEmpty() {
    * @return true if the timetable changed as a result of the call
    */
   private boolean clearTimetables(String feedId) {
-    return timetables.keySet().removeIf(tripPattern -> feedId.equals(tripPattern.getFeedId()));
+    var dirty = false;
+    dirtyTimetables.clear();
```

Comment:
I not sure if it's less confusing to just tell the TransitLayerUpdater to drop all previously added timetables. I think @vpaturet will be the best person to say.

Reply:
TransitLayerUpdater is not feed aware, and it also doesn't have keep the original timetable in the state, so it is not possible for me to do that.